### PR TITLE
mkcloud: extract VM and network creation into a separate script

### PR DESCRIPTION
**Pull request by MaximilianMeister**
This split-off would be a first step for refactoring

### CODE REVIEW

File: scripts/mkcloud [hunk 1]
```diff
@@ -15,6 +15,9 @@
 # 3. Please 'export' environment variables according to your needs.
 # 4. read the usage: mkcloud help
 
+# include separate scripts
+source $(dirname $0)/libvirt.sh
```
Comment (MaximilianMeister) [inline scripts/mkcloud]: check if $0 is a symlink, when symlink maybe use readlink

**Comment by aspiers** [comment]
Great, thanks!  If this is a straight extract of code with nothing changed then :+1:

**Comment by MaximilianMeister** [comment]
besides the include part, and adding a `libvirt_` prefix to function names, this is indeed just a straight extract. @bmwiedemann @jdsn are you fine with this?

**Comment by jdsn** [comment]
Thats a plain extract. Thats good as a first step to create a real separate and independent tool.
As the current code uses and relies on variable of mklcoud, I would not call it just "libvirt.sh". The name should rather show that it (still) depends on mkcloud. So I recommend "mkcloud-libvirt.sh".

Once this is refactored and it is a stand alone tool we can rename it to a general name.

File: scripts/mkcloud [hunk 2]
```diff
@@ -15,6 +15,19 @@
 # 3. Please 'export' environment variables according to your needs.
 # 4. read the usage: mkcloud help
 
+# include separate bash libs
+function include_bash_libs()
+{
+    local path=$(dirname $0)
+    local scripts="libvirt.sh"
```
Comment (jdsn) [inline scripts/mkcloud]: This is an invitation to only move code outside. Our goal is to create separate independent scripts that can be reused not only in mkcloud context.

No need to change the code here. But maybe a prominent comment might be good to point out that this mechanism is only a temporary solution during refactoring.

**Comment by jdsn** [comment]
:+1:

**Comment by MaximilianMeister** [comment]
just rebased

**Pull request merged**